h/views/api_auth.py

```diff
@@ -2,102 +2,88 @@
+    InvalidRequestFatalError,
+    WebApplicationServer,
+)
 from pyramid import security
+from pyramid.httpexceptions import (
+    HTTPFound,
+    HTTPInternalServerError,
+    exception_response,
+)
+from pyramid.view import view_config, view_defaults
 
+from h import models
 from h.exceptions import OAuthTokenError
-from h.util.view import cors_json_view
 from h.util.datetime import utc_iso8601
+from h.util.view import cors_json_view
+
+log = logging.getLogger(__name__)
 
 
 @view_defaults(route_name='oauth_authorize')
 class OAuthAuthorizeController(object):
-
     def __init__(self, request):
         self.request = request
-        self.oauth_svc = self.request.find_service(name='oauth')
         self.user_svc = self.request.find_service(name='user')
 
+        validator = self.request.find_service(name='oauth_validator')
+        self.oauth = WebApplicationServer(validator)
+
     @view_config(request_method='GET',
                  renderer='h:templates/oauth/authorize.html.jinja2')
     def get(self):
-        """
-        Check the user's authentication status and present the authorization
-        page.
-        """
-        self._check_params()
+        scopes, credentials = self.oauth.validate_authorization_request(self.request.url)
 
         if self.request.authenticated_userid is None:
             raise HTTPFound(self.request.route_url('login', _query={
                               'next': self.request.url}))
 
-        params = self.request.params
+        client_id = credentials.get('client_id')
+        state = credentials.get('state')
+
         user = self.user_svc.fetch(self.request.authenticated_userid)
+        client = self.request.db.query(models.AuthClient).get(client_id)
 
         return {'username': user.username,
-                'client_name': 'Hypothesis',
-                'client_id': params['client_id'],
-                'response_type': params['response_type'],
-                'response_mode': params['response_mode'],
-                'state': params.get('state')}
+                'client_name': client.name,
+                'client_id': client.id,
+                'response_type': client.response_type.value,
+                'state': state}
 
     @view_config(request_method='POST',
-                 renderer='h:templates/oauth/post_authorize.html.jinja2',
                  effective_principals=security.Authenticated)
     def post(self):
-        """
-        Process an authentication request and return an auth code to the client.
-
-        Depending on the "response_mode" parameter the auth code will be
-        delivered either via a redirect or via a `postMessage` call to the
-        opening window.
-        """
-        authclient = self._check_params()
-
+        # We don't support scopes at the moment, but oauthlib does need a scope,
+        # so we're adding a hard-coded `annotation:read` and `annotation:write` scopes.
+        scopes = ['annotation:read', 'annotation:write']
```

Comment:
Is this different from the return value of `get_default_scopes` in the request validator? Is there a way we can define default scopes just once?